pkgs/development/python-modules/marimo/default.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, setuptools
+, click
+, jedi
+, markdown
+, pymdown-extensions
+, pygments
+, tomlkit
+, uvicorn
+, starlette
+, websockets
+, docutils
+, black
+}:
+
+buildPythonPackage rec {
+  pname = "marimo";
+  version = "0.3.3";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-l5QehU/LqEWb7ybKxace4sm6C9mcNlaHNOp55ExglpQ=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    click
+    jedi
+    markdown
+    pymdown-extensions
+    pygments
+    tomlkit
+    uvicorn
+    starlette
+    websockets
+    docutils
+    black
+  ];
+
+  meta = with lib; {
+    description = "A reactive notebook for Python — run reproducible experiments, execute as a script, deploy as an app, and version with git.";
+    homepage = "https://github.com/marimo-team/marimo";
+    license = licenses.asl20;
```

```suggestion
    license = licenses.asl20;
    mainProgram = "marimo";
```
If an app has an obvious "main executable" then it should be named in `meta.mainProgram`